src/test/java/com/amazonaws/encryptionsdk/kmsv2/XCompatKmsDecryptTest.java

```diff
@@ -3,29 +3,27 @@
 
 package com.amazonaws.encryptionsdk.kmsv2;
 
+import static org.junit.Assert.assertArrayEquals;
+
 import com.amazonaws.encryptionsdk.AwsCrypto;
 import com.amazonaws.encryptionsdk.CryptoResult;
-import com.amazonaws.encryptionsdk.kms.KmsMasterKeyProvider;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
 import org.apache.commons.lang3.StringUtils;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 import org.junit.runners.Parameterized.Parameters;
 
-import java.io.File;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.*;
-
-import static org.junit.Assert.assertArrayEquals;
-
 @RunWith(Parameterized.class)
 public class XCompatKmsDecryptTest {
-  private String plaintextFileName;
-  private String ciphertextFileName;
-  private String kmsKeyId;
+  private final String plaintextFileName;
+  private final String ciphertextFileName;
+  private final String kmsKeyId;
 
   public XCompatKmsDecryptTest(
```

Comment:
We should add this (and MaxEncryptedDataKeysIntegrationTest.java) to https://github.com/aws/aws-encryption-sdk-java/blob/275ce526953ecf8f8f2f94ba38dcdf92e35c84bf/src/test/java/com/amazonaws/encryptionsdk/IntegrationTestSuite.java

Reply:
That being said, I don't think we run IntegrationTestSuite anywhere in our CI, it doesn't even look like the pom defines any target for it... Historically, I think these tests were turned off before my time on the team because they just took too long. It might be worth taking another look now to determine whether we think there is actually any value for defining KMSSDKv2 versions of these tests, whether it's worth adding these tests at some point in our CI (maybe just a daily run or at least add to the release process?), or whether we should clean up the state of this repo and remove these tests if they are not in use.

Reply:
I'll add them here, but yea, they're not specified in any target. I've run the test suites manually to make sure they pass though.